test/ui-testing/pick-lists.js

```diff
@@ -0,0 +1,268 @@
+
+module.exports.test = (uiTestCtx) => {
+  const number = Math.round(Math.random() * 100000);
+  const editText = '-edited';
+  const pickList = `${number}pickList`;
+  const editedPickList = `${pickList}${editText}`;
+  const pickListValue = `${number}pickListValue`;
+  const editedPickListValue = `${pickListValue}${editText}`;
+
+  describe('Pick list crud', function test() {
+    const { config, helpers } = uiTestCtx;
+    const nightmare = new Nightmare(config.nightmare);
+    nightmare.options.width = 1300; // added this temporarily as MultiSelect doesnt work well with narrow screen sizes
+
+    this.timeout(Number(config.test_timeout));
+
+    describe('Login > Open settings > Create and edit pick list > Create and edit pick list value > Delete value and pick list > Logout\n', () => {
+      before((done) => {
+        helpers.login(nightmare, config, done);
+      });
+
+      after((done) => {
+        helpers.logout(nightmare, config, done);
+      });
+
+      it('should open Settings', done => {
+        helpers.clickSettings(nightmare, done);
+      });
+
+      it(`should create pick list ${pickList} in settings`, done => {
+        nightmare
+          .wait('a[href="/settings/licenses"]')
+          .click('a[href="/settings/licenses"]')
+          .wait('a[href="/settings/licenses/pick-lists"]')
+          .click('a[href="/settings/licenses/pick-lists"]')
+          .waitUntilNetworkIdle(1000)
+          .wait('#clickable-add-pick-lists')
+          .click('#clickable-add-pick-lists')
+          .wait('input[name="items[0].desc"]')
+          .type('input[name="items[0].desc"]', pickList)
+          .wait('#clickable-save-pick-lists-0')
+          .click('#clickable-save-pick-lists-0')
+          .wait(222)
+          .then(done)
+          .catch(done);
+      });
+
+      it(`should find pick list ${pickList}`, done => {
+        nightmare
+          .wait('a[href="/settings/licenses"]')
+          .click('a[href="/settings/licenses"]')
+          .wait('a[href="/settings/licenses/pick-lists"]')
+          .click('a[href="/settings/licenses/pick-lists"]')
+          .wait('#editList-pick-lists')
+          .evaluate(list => {
+            const listRows = [...document.querySelectorAll('#editList-pick-lists')].map(e => e.textContent);
+            const listElement = listRows.find(r => r.indexOf(list) >= 0);
+            if (!listElement) {
+              throw Error(`Could not find row with the list ${list} but found ${listElement}`);
+            }
+          }, pickList)
+          .then(done)
+          .catch(done);
+      });
+
+      it(`should edit pick list ${pickList} in settings`, done => {
+        nightmare
+          .wait('a[href="/settings/licenses"]')
+          .click('a[href="/settings/licenses"]')
+          .wait('a[href="/settings/licenses/pick-lists"]')
+          .click('a[href="/settings/licenses/pick-lists"]')
+          .waitUntilNetworkIdle(1000)
+          .wait('#clickable-edit-pick-lists-0')
```

Comment:
This is unsafe since you're assuming that the first item will be the item that was just created, which isn't a safe assumption. Since you're blindly editing the name of the first item in the list you can corrupt existing data.

Instead, use an `evaluate` to figure out which row contains the pick list you just created, `return` the row number from it, use that return value in a `then` where you use it.

Eg,
```
.evaluate(_pickList => {
  const rowNumber = // figure out the row number
  return rowNumber;
}, pickList)
.then(rowNumber => {
  nightmare
    .click(`#clickable-edit-pick-lists-${rowNumber}`)
    .wait(...)
    .insert(...)
    .then(done)
    .catch(done);
})
.catch(done);
```